content/blog/2019/egress-traffic-control-in-istio-part-3/index.md

```diff
@@ -0,0 +1,139 @@
+securing egress traffic and showed how you can prevent the attacks using Istio.
+
+In this installment, I compare secure control of egress traffic in Istio with alternative solutions such as Kubernetes
+Network Policies and legacy egress proxies/firewalls. Finally, I describe the performance considerations regarding the
+secure control of egress traffic in Istio.
+
+## Alternative solutions for egress traffic control
+
+First, let me remind you the [requirements for egress traffic control we collected](/blog/2019/egress-traffic-control-in-istio-part-1/#requirements-for-egress-traffic-control):
+
+1.  Support for [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) with
+    [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) or for [TLS origination](/docs/reference/glossary/#tls-origination) by Istio.
+1.  **Monitor** SNI and the source workload of every egress access.
+1.  Define and enforce **policies per cluster**.
+1.  Define and enforce **policies per source**, _Kubernetes-aware_.
+1.  **Prevent tampering**.
+1.  Traffic control is **transparent** to the applications.
+
+Now, let me present two alternative solutions for egress traffic control and show which requirements they satisfy, and,
+more importantly, cannot satisfy.
+
+The most natural solution for egress traffic control is
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/). Using
+Kubernetes Network Policies, cluster operators can specify which external services can be accessed by which pods. The
+pods may be identified by pod labels, namespace labels, or by IP ranges. The external services can be specified by IP
+ranges: Kubernetes Network Policies are not DNS-aware. The first requirement is satisfied since any TCP traffic can be
+controlled by Kubernetes Network Policies. The requirements 3 and 4 are satisfied partially: the policies can be
+specified per cluster or per pod, however the external services cannot be identified by domain names.
+The fifth requirement is satisfied if the attackers are not able to break from a malicious container into the Kubernetes
+node and to interfere with the implementation of the Kubernetes Network Policies inside the node.
+The sixth requirement is satisfied as well: there is no need to change the code or the
+container environment. We can say that Kubernetes Network Policies provide transparent, Kubernetes-aware egress traffic
+control, which is not DNS-aware.
+
+Another approach that predates Kubernetes network policies is a **DNS-aware egress proxy/firewall**. In this
+approach, applications are configured to direct the traffic to the proxy and to use some proxy protocol, e.g.
+[SOCKS](https://en.wikipedia.org/wiki/SOCKS).
+Since the applications must be specially configured, this solution is not transparent. Moreover, egress proxies are not
+Kubernetes-aware, since neither pod labels nor pod service account are known to the egress proxy. Such egress proxies
```

Comment:
@rcaballeromx Good idea to use operators as the subjects. Good that you did not use **cluster** operators, since the egress proxies are not a Kubernetes concept, so Kubernetes cluster operators not necessarily configure them, some other operators do.